flow/connectors/postgres/cdc.go

```diff
@@ -306,7 +311,49 @@ func (p *PostgresCDCSource) decodeColumnData(
 ) (types.QValue, error) {
 	var parsedData any
 	var err error
-	if dt, ok := p.typeMap.TypeForOID(dataType); ok {
+
+	// Special handling for JSON types to use relaxed number parsing
+	if dataType == pgtype.JSONOID || dataType == pgtype.JSONBOID {
+		jsonData := data
+		// For JSONB in binary format, skip the version byte
+		if formatCode == pgtype.BinaryFormatCode {
+			if dataType == pgtype.JSONBOID {
+				if len(data) == 0 {
+					return nil, fmt.Errorf("jsonb too short")
+				}
+				if data[0] != 1 {
+					return nil, fmt.Errorf("unknown jsonb version number %d", data[0])
+				}
+				jsonData = data[1:]
+			}
+		} else if formatCode != pgtype.TextFormatCode {
+			return nil, fmt.Errorf("unknown format code for json: %d", formatCode)
+		}
+
+		if err := p.jsonApi.Unmarshal(jsonData, &parsedData); err != nil {
+			p.logger.Error("[pg_cdc] failed to unmarshal json", slog.Any("error", err))
+			return nil, fmt.Errorf("failed to unmarshal json: %w", err)
+		}
+		return p.parseFieldFromPostgresOID(dataType, typmod, parsedData, customTypeMapping, p.internalVersion)
```

Comment:
nit: using the same approach as how JSONArrayOID/JSONBArrayOID are handled,  could this be simplified to 
```
	if dataType == pgtype.JSONOID || dataType == pgtype.JSONBOID {
		var text pgtype.Text
		if err := p.typeMap.Scan(dataType, formatCode, data, text); err != nil {
			return nil, fmt.Errorf("error decoding binary column data: %w", err)
		}
		if err := p.jsonApi.UnmarshalFromString(text.String, &parsedData); err != nil {
			p.logger.Error("[pg_cdc] failed to unmarshal json", slog.Any("error", err))
			return nil, fmt.Errorf("failed to unmarshal json: %w", err)
		}
		return p.parseFieldFromPostgresOID(dataType, typmod, parsedData, customTypeMapping, p.internalVersion)
```
which avoid needing to know about encodings (albeit it being a very simple one)